internal/resource/obtenant/utils.go

```diff
@@ -920,3 +922,91 @@ func CreateUserWithCredentials(m *OBTenantManager) error {
 	}
 	return nil
 }
+
+func (m *OBTenantManager) listOBParameters() (*v1alpha1.OBParameterList, error) {
+	obparameterList := &v1alpha1.OBParameterList{}
+	err := m.Client.List(m.Ctx, obparameterList, client.MatchingLabels{
+		oceanbaseconst.LabelRefOBCluster: m.OBTenant.Spec.ClusterName,
+	}, client.InNamespace(m.OBTenant.Namespace))
+	if err != nil {
+		return nil, errors.Wrap(err, "get obparameter list")
+	}
+	return obparameterList, nil
+}
+
+func (m *OBTenantManager) generateParameterName(name string) string {
+	return fmt.Sprintf("%s-%s-%s", m.OBTenant.Spec.ClusterName, m.OBTenant.Name, strings.ReplaceAll(name, "_", "-"))
+}
+
+func (m *OBTenantManager) createOBParameter(parameter *apitypes.Parameter) error {
+	m.Logger.Info("Create ob tenant parameters")
+	ownerReferenceList := make([]metav1.OwnerReference, 0)
+	ownerReference := metav1.OwnerReference{
+		APIVersion: m.OBTenant.APIVersion,
+		Kind:       m.OBTenant.Kind,
+		Name:       m.OBTenant.Name,
+		UID:        m.OBTenant.GetUID(),
+	}
+	ownerReferenceList = append(ownerReferenceList, ownerReference)
+	labels := make(map[string]string)
+	labels[oceanbaseconst.LabelRefUID] = string(m.OBTenant.GetUID())
+	labels[oceanbaseconst.LabelRefOBCluster] = m.OBTenant.Spec.ClusterName
+	parameterName := m.generateParameterName(parameter.Name)
+	obparameter := &v1alpha1.OBParameter{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            parameterName,
+			Namespace:       m.OBTenant.Namespace,
+			OwnerReferences: ownerReferenceList,
+			Labels:          labels,
+		},
+		Spec: v1alpha1.OBParameterSpec{
+			ClusterName: m.OBTenant.Spec.ClusterName,
+			TenantName:  m.OBTenant.Spec.TenantName,
+			TenantId:    int64(m.OBTenant.Status.TenantRecordInfo.TenantID),
+			Parameter:   parameter,
+		},
+	}
+	m.Logger.V(oceanbaseconst.LogLevelDebug).Info("Create obparameter", "parameter", parameterName)
+	err := m.Client.Create(m.Ctx, obparameter)
+	if err != nil {
+		m.Logger.Error(err, "create obparameter failed")
+		return errors.Wrap(err, "create obparameter")
+	}
+	return nil
+}
+
+func (m *OBTenantManager) updateOBParameter(parameter *apitypes.Parameter) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
`RetryOnConflict` will take effect only when the function get an `IsConflict` error which is defined in client-go packages. Maybe we should use more general `OnError` here.

Reply:
I think using RetryOnConflict is the right choice, this function is called by tenant manager and update the resource parameter managed by paramenter manager, it's possible to encounter a conflict scenario, if there's other error, we can leave that in another reconcile loop.

Reply:
Oh sorry, it's right. I thought there are direct SQL statement execution in the function.